packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -783,8 +800,10 @@ export default function parse(source: string, state: State): TemplateParseResult
                 return warnOnElement(ParserDiagnostics.ROOT_TAG_SHOULD_BE_TEMPLATE, node, [tag]);
             }
 
-            const hasAttributes = node.attrs.length !== 0;
-            if (hasAttributes) {
+            const rootHasUnknownAttributes = node.attrs.some(
+                ({ name }) => name !== PRESERVE_COMMENTS_OPTION
+            );
+            if (rootHasUnknownAttributes) {
                 return warnOnElement(ParserDiagnostics.ROOT_TEMPLATE_CANNOT_HAVE_ATTRIBUTES, node);
```

Comment:
The error message and error name should probably be updated now that we allow attributes on the root `<template>` element.